confidant/authnz/userauth.py

```diff
@@ -299,22 +319,27 @@ def assert_headers(self):
     def current_user(self):
         self.assert_headers()
 
-        info = {
-            'email': request.headers[self.email_header],
-
-            # TODO: should we use a string like "unknown", fall back to the
-            # email/username, ...?
-            'first_name': '',
-            'last_name': '',
-        }
-
+        first_name = None
         if self.first_name_header and self.first_name_header in request.headers:
-            info['first_name'] = request.headers[self.first_name_header]
+            first_name = request.headers[self.first_name_header]
 
+        last_name = None
         if self.last_name_header and self.last_name_header in request.headers:
-            info['last_name'] = request.headers[self.last_name_header]
-
-        return info
+            last_name = request.headers[self.last_name_header]
+
+        username = None
+        if self.username_header and self.username_header in request.headers:
+            username = request.headers[self.username_header]
+
+        # Set current user on every request to ensure that the user's group
+        # list is updated (TODO: caching?)
+        self.set_current_user(
+            request.headers[self.email_header],
+            first_name = first_name,
+            last_name = last_name,
+            username = username
```

Comment:
first_name and last_name here are changing their defaults from '' to None. Will this cause typing issues anywhere?